### DISCUSSION TITLE
How to env file in compose.upOne or any compose operation.

### DISCUSSION THREAD

**Discussion by dishabodade** [PDMLab/docker-compose#183] [Q&A]
I have env file which we can use as  
$docker-compose --env-file declaration.env up -d
$docker-compose --env-file declaration.env down

I have implemented some API's using node docker-compose package.
I am using function call like  
```
 compose.upOne('nginx', { cwd: arsAppDir, log: true }).then(() => {
                        console.log('done');
                        resolve(0);
                        }, (error: any) => {
                             resolve(1);
                        });
```

how to add this option --env-file declaration.env?
I have added the below way.
```
 compose.upOne('nginx', { cwd: arsAppDir, log: true }, ["--env-file", "declaration.env"] ).then(() => {
                        console.log('done');
                        resolve(0);
                        }, (error: any) => {
                             resolve(1);
                        });

```
I am doing it right?

**Comment by Steveb-p** [comment]
Unfortunately that won't work afaik. `upOne` does not have a third argument:

```typescript
export const upOne = function (
  service: string,
  options?: IDockerComposeOptions
): Promise<IDockerComposeResult> {
  const args = shouldUseDefaultNonInteractiveFlag(options)
    ? ['-d', service]
    : [service]
  return execCompose('up', args, options)
}
```

What you want to use is the `options.composeOptions` argument:
https://github.com/PDMLab/docker-compose/blob/a28e8015903db9b8f401e72dde7a74d4f50fd5b2/src/index.ts#L11

Like so (haven't tested this):
```typescript
compose.upOne('nginx', { cwd: arsAppDir, log: true, composeOptions: ['--env-file', 'declarations.env'] })
```

**Reply by dishabodade** [reply]
yes, its working with the composeOptions.

One more issue I am facing while using docker-compose package.

As per the docker-compose up -d behavior, if we do docker-compose up -d it will recreate the services which has changes.
but with npm package, its restarting depends_on services also.

docker-compose version I am using is 
docker-compose version 1.27.4

Any setting or suggestions, how can i handle this case. I don't want to restart all containers on compose.upOne or compose.upAll
